CAMEL-10966 Salesforce Maven Plugin Escape Strings

NickCis opened the pull request:
Salesforce Maven Plugin doesn't escape strings when doing the `camel-salesforce:generate` phase.

This PR uses the _velocity escape tool_ in order to escape strings.

davsclaus commented:
Thanks

@zregvart do you mind taking a look at this PR ?

zregvart commented:
@NickCis thank you for this and good catch! Can you please log a JIRA issue for this at https://issues.apache.org/jira/browse/CAMEL/ (just mark it as a bug in 2.19 and against camel-salesforce component), and then squash these commits and amend the git commit message with the JIRA number?

Something along the lines of:

    $ git rebase -i HEAD~2

set `squash` to the second commit, change the commit message to `CAMEL-XXXX Salesforce Maven Plugin Escape Strings` and force push the branch to your fork (`git push -f`).

Thank you :+1:

NickCis commented:
@zregvart Done.
You're welcomed, i'm happy to contribute to the project! :)

zregvart commented:
@NickCis thank you for your contribution it has been merged, you can close this pull request. And if you find any other issues please don't hesitate to create more pull requests, we :heart:  them :+1:

NickCis commented:
@zregvart I've found that there was missing the escaping in the `sobject-pojo` file. I've added it in the last commit.

In addition, i've found a bug in the way the velocity escapes. If there is a slash `/`, velocity escapes it to `\/` which is invalid. I think this is due to a _apache lang2 EscapeUtils_ bug (which is what _Velocity Escape tools_ uses), i've tested using _apache lang3 EscapeUtils_ directly and the escaping works correctly.

Should i create another Jira ticket and send another PR? (I've fixed it in [this branch](https://github.com/nickcis/camel/tree/salesforce-enum-escape)).

Is there a recommended way in order to write some tests for this specific case?

zregvart commented:
@NickCis thanks for testing this I hadn't had the chance to test it, we can re-open the ticket and you can do another pull request.
Testing Maven plugins is doable, but I wonder if you can do a test with just the velocity templates that might prove to be a bit simpler.
Open a PR with what you have and if you get into pains testing this we collaborate on a test together.

NickCis commented:
I've opened another [pr](https://github.com/apache/camel/pull/1526) to address this issues.
